utils/preprocess.py

```diff
@@ -1,40 +1,42 @@
-from typing import List
-
-from PIL.Image import Image
+from typing import List, Optional, Tuple, Union
 import numpy as np
 from torch.utils.data import Dataset
 import random
 from tqdm import tqdm
 import torch
-
-
-class SIEDataset(Dataset):
-    # load all images at once
-    # we use float32 for the image data type
-    def __init__(self, sand_dust_images: List[Image], ground_truth_images: List[Image], image_names: List[str], w_threshold: int, h_threshold: int):
-        assert len(sand_dust_images) == len(ground_truth_images), "the number of sand dust images and ground truth images must be the same"
+import os
+from pathlib import Path
+from PIL import Image
+
+class ImageDataset(Dataset):
+    W_THRESHOLD = 440
+    H_THRESHOLD = 330
+    
+    def __init__(self, sand_dust_images: List[Image.Image], image_names: List[str], ground_truth_images: Optional[List[Image.Image]] = None):
         self.sand_dust_images = sand_dust_images
         self.ground_truth_images = ground_truth_images
         self.image_names = image_names
-        self.w_threshold = w_threshold
-        self.h_threshold = h_threshold
+        self.is_paired = ground_truth_images is not None
 
     def __len__(self):
         return len(self.sand_dust_images)
 
     def preprocess(self, image: Image) -> torch.Tensor:
-        image = resize_image(image, self.w_threshold, self.h_threshold)
+        image = resize_image(image, self.W_THRESHOLD, self.H_THRESHOLD)
```

Comment:
I think this code does not work? should be ImageDataset.W_THRESHOLD and ImageDataset.H_THRESHOLD  because you are refering to the class variable

Reply:
<img width="1470" alt="Screenshot 2024-03-30 at 11 07 37" src="https://github.com/superleesa/final_year_project/assets/144698049/52f69154-b7ec-4a3d-bec8-10bd3210ce0f">
It's inside the class so I thought that we can call the class variable like that? I've attached the screenshot.

Reply:
The code itself works as well. Will it occur logical error or something?

Reply:
yeah it seems to be working so it's fine